dash/include/dash/algorithm/ForEach.h

```diff
@@ -96,25 +58,27 @@ void for_each(
  * function on its local elements only. The index passed to the function is
  * a global index.
  *
- * \tparam      ElementType  Type of the elements in the sequence
- * \tparam      IndexType    Parameter type expected by function to
- *                           invoke, deduced from parameter \c func
+ * \tparam      ElementType            Type of the elements in the sequence
+ * \tparam      UnaryFunctionWithIndex Function to invoke for each element
+ *                                     in the specified range with signature
+ *                                     \c void (const ElementType &, index_t)
```

Comment:
Does not "have to", but it should:

<http://en.cppreference.com/w/cpp/algorithm/for_each>

The qualifier merely states that accepting const references to values is supported.